include/inviwo/core/datastructures/geometry/typedmesh.h

```diff
@@ -0,0 +1,489 @@
+    using Base::Base;
+
+    std::shared_ptr<const Buffer<type>> getNormals() const { return Base::buffer_; }
+    std::shared_ptr<Buffer<type>> getEditableNormals() { return Base::buffer_; }
+
+    void setVertexNormal(size_t index, vec3 normal){
+        getEditableNormals()->getEditableRAMRepresentation()->set(index,normal);
+    }
+};
+
+/**
+ * \ingroup typedmesh
+ * BufferTrait for Colors buffers (glm::vec4)
+ */
+class ColorsBuffer : public BufferTrait<float, 4, static_cast<int>(BufferType::ColorAttrib)> {
+public:
+    using Base = BufferTrait<float, 4, static_cast<int>(BufferType::ColorAttrib)>;
+    using Base::Base;
+
+    std::shared_ptr<const Buffer<type>> getColors() const { return Base::buffer_; }
+    std::shared_ptr<Buffer<type>> getEditableColors() { return Base::buffer_; }
+
+    void setVertexColor(size_t index, vec4 color){
+        getEditableColors()->getEditableRAMRepresentation()->set(index,color);
+    }
+};
+
+/**
+ * \ingroup typedmesh
+ * BufferTrait for Texture Coordinate buffers (glm::vec3)
+ */
+class TexcoordBuffer : public BufferTrait<float, 3, static_cast<int>(BufferType::TexcoordAttrib)> {
+public:
+    using Base = BufferTrait<float, 3, static_cast<int>(BufferType::TexcoordAttrib)>;
+    using Base::Base;
+
+    std::shared_ptr<const Buffer<type>> getTexCoords() const { return Base::buffer_; }
+    std::shared_ptr<Buffer<type>> getEditableTexCoords() { return Base::buffer_; }
+
+    void setVertexTexCoord(size_t index, vec3 texCoord){
+        getEditableTexCoords()->getEditableRAMRepresentation()->set(index,texCoord);
+    }
+};
+
+/**
+ * \ingroup typedmesh
+ * BufferTrait for Curvature buffers (float)
+ */
+class CurvatureBuffer
+    : public BufferTrait<float, 1, static_cast<int>(BufferType::CurvatureAttrib)> {
+public:
+    using Base = BufferTrait<float, 1, static_cast<int>(BufferType::CurvatureAttrib)>;
+    using Base::Base;
+
+    std::shared_ptr<const Buffer<type>> getCurvatures() const { return Base::buffer_; }
+    std::shared_ptr<Buffer<type>> getEditableCurvatures() { return Base::buffer_; }
+
+    void setVertexCurvature(size_t index, float curvature){
+        getEditableCurvatures()->getEditableRAMRepresentation()->set(index,curvature);
+    }
+};
+
+/**
+ * \ingroup typedmesh
+ * BufferTrait for Uint32 buffers
+ */
+using IndexBuffer = BufferTrait<uint32_t, 1, static_cast<int>(BufferType::IndexAttrib)>;
+
+/**
+ * \ingroup typedmesh
+ * BufferTrait for radii buffers (float)
+ * \see SphereMesh
+ */
+class RadiiBuffer
+    : public BufferTrait<float, 1, static_cast<int>(BufferType::NumberOfBufferTypes)> {
+public:
+    using Base = BufferTrait<float, 1, static_cast<int>(BufferType::NumberOfBufferTypes)>;
+    using Base::Base;
+
+    std::shared_ptr<const Buffer<type>> getRadii() const { return Base::buffer_; }
+    std::shared_ptr<Buffer<type>> getEditableRadii() { return Base::buffer_; }
+
+    void setVertexRadius(size_t index, float radius){
```

Comment:
rename to `setRadius` to be consistent

Reply:
Consitent with what? The other BufferBase-classes has setVertexPosition or setVertexNormal etc. That is to mimic the interface of basic mesh.